### PR TITLE
WIP: Implementation of Topics.

### PR DESCRIPTION
**Pull request by caneppelevitor**
image
image
image

### CODE REVIEW

File: server/topic/topic.controller.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { TopicService } from "./topic.service";
+
+@Controller()
+export class TopicController {
+    private readonly logger = new Logger("");
+    constructor(private topicService: TopicService) {}
+
+    @Post("api/topics")
+    async create(@Body() topics) {
+        console.log("first");
```
Comment (thesocialdev) [inline server/topic/topic.controller.ts:19]: Remove console.log

File: server/topic/topic.controller.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    Logger,
+    Param,
+    Post,
+    Query,
+} from "@nestjs/common";
+import { TopicService } from "./topic.service";
+
+@Controller()
+export class TopicController {
+    private readonly logger = new Logger("");
```
Comment (thesocialdev) [inline server/topic/topic.controller.ts:14]: Add a string to the Logger instatiation

File: server/topic/topic.controller.ts (hunk 1 above)
Comment (caneppelevitor) [reply server/topic/topic.controller.ts:19]: Sorry, it should be a draft PR

File: server/topic/topic.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export class TopicService {
+    constructor(
+        @InjectModel(Topic.name)
+        private TopicModel: Model<TopicDocument>
+    ) {}
+
+    async create(topics) {
+        await Promise.all(
+            topics.map(async (topic) => {
+                console.log(topic, "Topic");
```
Comment (thesocialdev) [inline server/topic/topic.service.ts:16]: Remove console.log

File: src/components/ClaimReview/TopicInput.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Col } from "antd";
+import { useState } from "react";
+import api from "../../api/topicsApi";
+import AletheiaButton from "../Button";
+import UserAutocomplete from "./UserAutocomplete";
+
+const TopicInput = ({}) => {
+    const [topics, setTopics] = useState<any>([]);
+
+    console.log(topics);
```
Comment (thesocialdev) [inline src/components/ClaimReview/TopicInput.tsx:10]: Remove console.log

File: src/components/ClaimReview/TopicInput.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    return (
+        <Col>
+            <UserAutocomplete
```
Comment (thesocialdev) [inline src/components/ClaimReview/TopicInput.tsx:13]: Should we rename the component to just `Autocomplete`?

File: src/components/ClaimReview/UserAutocomplete.tsx [hunk 6]
```diff
@@ -1,28 +1,39 @@
...
-import { useTranslation } from 'next-i18next'
-import SelectUser from './form/SelectUser'
+import { useTranslation } from "next-i18next";
+import SelectUser from "./form/SelectUser";
 
-const UserAutocomplete = ({ placeholder, onChange, dataCy }) => {
+const UserAutocomplete = ({ placeholder, onChange, dataCy, fieldName }) => {
     const { t } = useTranslation();
 
     const fetchUserList = async (name) => {
```
Comment (thesocialdev) [inline src/components/ClaimReview/UserAutocomplete.tsx:9]: Renaming the component to Autocomplete means that we should pass the `fetchUserList` or the `fetchTopicList` as "data loaders", adding a new prop to the component `dataLoader`

File: src/components/ClaimReview/form/SelectUser.tsx [hunk 7]
```diff
@@ -31,7 +31,7 @@ function SelectUser({ fetchOptions, ...props }) {
 
     return (
         <StyledSelect
-            mode="multiple"
+            mode={name === "usersId" ? "multiple" : "tags"}
```
Comment (thesocialdev) [inline src/components/ClaimReview/form/SelectUser.tsx:34]: pass styled as a prop instead, the component shouldn't care about it's being for a user or a topic

**Review by thesocialdev — CHANGES_REQUESTED** [review]
- [ ] Change the gear icon to a pencil icon, to indicate that the topics can be edited
- [ ] Change the label "Edit Topics" to "Add Topics"

Question, how do new topics are inserted in the platform?

**Comment by pepermao** [comment]
> * [x]  Change the gear icon to a pencil icon, to indicate that the topics can be edited
> * [x]  Change the label "Edit Topics" to "Add Topics"
> 
> Question, how do new topics are inserted in the platform?

I documented it in the topic.service code

**Review by thesocialdev — APPROVED** [review]

**Pull request merged**